README.md

```diff
@@ -4,28 +4,149 @@ Mediawiki title normalizetion, that conforms to the normalization rules used in
+        * [.main(siteInfo)](#Namespace.main) ⇒ <code>[Namespace](#Namespace)</code>
+
+<a name="new_Namespace_new"></a>
+### new Namespace(id, siteInfo)
+Represents a wiki namespace
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| id | <code>number</code> | The namespace identifier |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | The site metadata information. |
+
+<a name="Namespace+isSpecial"></a>
+### namespace.isSpecial() ⇒ <code>boolean</code>
+Checks whether namespace is `Special`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isMain"></a>
+### namespace.isMain() ⇒ <code>boolean</code>
+Checks whether namespace is `Main`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isTalk"></a>
+### namespace.isTalk() ⇒ <code>boolean</code>
+Checks whether namespace is `Talk`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isUser"></a>
+### namespace.isUser() ⇒ <code>boolean</code>
+Checks whether namespace is `User`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+isUserTalk"></a>
+### namespace.isUserTalk() ⇒ <code>boolean</code>
+Checks whether namespace is `User_Talk`.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+getNormalizedText"></a>
+### namespace.getNormalizedText() ⇒ <code>string</code>
+Get the canonical name string for this namespace.
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace+getCase"></a>
+### namespace.getCase() ⇒ <code>string</code>
+Returns the case parameter for the namespace
+
+**Kind**: instance method of <code>[Namespace](#Namespace)</code>  
+<a name="Namespace.fromText"></a>
+### Namespace.fromText(text, siteInfo) ⇒ <code>[Namespace](#Namespace)</code> &#124; <code>undefined</code>
+Creates a namespace instance from namespace text or a namespace alias
+
+**Kind**: static method of <code>[Namespace](#Namespace)</code>  
+**Returns**: <code>[Namespace](#Namespace)</code> &#124; <code>undefined</code> - a namespace or undefined if it wasn't found.  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| text | <code>string</code> | Namespace name text. |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | the site information. |
+
+<a name="Namespace.main"></a>
+### Namespace.main(siteInfo) ⇒ <code>[Namespace](#Namespace)</code>
+Creates a namespace object for a `Main` namespace.
+
+**Kind**: static method of <code>[Namespace](#Namespace)</code>  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| siteInfo | <code>[SiteInfo](#SiteInfo)</code> | the site information. |
+
+<a name="Title"></a>
+## Title
+**Kind**: global class  
+
+* [Title](#Title)
+    * [new Title(key, namespace, siteInfo, [fragment])](#new_Title_new)
+    * _instance_
+        * [.getNormalizedText()](#Title+getNormalizedText) ⇒ <code>string</code>
```

Comment:
The MediaWiki implementation calls this [getPrefixedDBKey](https://github.com/wikimedia/mediawiki/blob/757714bf95c15342be0447e9b7d0b5c778a1f0f1/includes/Title.php#L1465). Do you think it would be worth using the same terminology?